src/main/java/org/rnorth/ansi/CapabilityDetection.java

```diff
@@ -0,0 +1,46 @@
+package org.rnorth.ansi;
+
+import com.sun.jna.Platform;
+
+/**
+ * Simple terminal capability detection.
+ */
+public class CapabilityDetection {
+    private static final int STDOUT_FILENO = 1;
+
+    private CapabilityDetection() {
+        // Utility class - don't allow construction;
+    }
+
+    public static boolean isUnderIDEA() {
+        // Rough check for whether or not we're running in IDEA
+        String classPath = System.getProperty("java.class.path");
+        return classPath.contains("idea_rt.jar");
+    }
+
+    public static boolean isUnderMaven() {
+        try {
+            // Rough check for whether or not we're running in a Maven build
+            Class.forName("org.codehaus.plexus.classworlds.launcher.Launcher");
+            return true;
+        } catch (Exception ignored) {
```

Comment:
`catch (Exception` is entirely wrong here, and same in `isTty` 😱 . Will fix.

Reply:
`catch` is fine, at least here, it handles it correctly, the stacktrace I sent I got from adding `e.printStackTrace()` :)